brion/plugin/compartmentReportBinary.cpp

```diff
@@ -666,29 +680,25 @@ bool CompartmentReportBinary::_parseMapping()
     perCellOffsets.resize(cells.size());
     _perCellCounts.resize(cells.size());
 
-    // According to Garik Suess, all compartments of a cell in a frame are next
-    // to each other, and all compartments of a section are next to each other.
-    // However, the sections are not necessarily sorted by their index in the
-    // frame. It could be that for cell with GID 50 while all data is
-    // contiguous, the sections are out of order so compartments for section 3 6
-    // 22 8 could be next to each other, while the compartments inside these
-    // sections are in order.
-    size_t idx = 0;
-    size_t totalCompartments = 0;
-    for (const CellInfo& info : cells)
+// With 8 threads there's some speed up, but very small. We stick with 6
+// to avoid using more than 1 socket.
+#pragma omp parallel for num_threads(6)
+    for (size_t idx = 0; idx < cells.size(); ++idx)
     {
+        CellInfo& info = cells[idx];
         uint16_t current = LB_UNDEFINED_UINT16;
         uint16_t previous = LB_UNDEFINED_UINT16;
         uint16_t count = 0;
 
         // < sectionID, < frameIndex, numCompartments > >
-        typedef std::map<uint16_t, std::pair<uint64_t, uint16_t>>
+        typedef std::vector<std::pair<uint16_t, std::pair<uint64_t, uint16_t>>>
             SectionMapping;
+
         SectionMapping sectionsMapping;
+        sectionsMapping.reserve(info.numCompartments);
 
         _perCellCounts[idx] = info.numCompartments;
-        perCellOffsets[idx] = totalCompartments;
-        totalCompartments += info.numCompartments;
+        perCellOffsets[idx] = info.accumCompartments;
 
         for (int32_t j = 0; j < info.numCompartments; ++j)
         {
```

Comment:
Maybe using the same datatype makes sense in reporting lib... It you're really concerned about it, please create a ticket so we don't forget. I'll add an assert for the moment, because changing uint16_t to uint32_t impacts the public API and the wrapping, not just this file.